### PR TITLE
Shared Config Credentials Tests

### PR DESCRIPTION
**Pull request by skmcgrail**
Adds the previously removed shared config credentials tests that were present in the v0.24.0 release. Contains various other minor fixes.

Requires https://github.com/awslabs/smithy-go/pull/186 to be approved and merged.

### CODE REVIEW

File: config/resolve_assume_role_test.go [hunk 1]
```diff
@@ -0,0 +1,201 @@
...
+		t.Errorf("expect %v, to be in %v", e, a)
+	}
+}
+
+func TestAssumeRole_WithMFA(t *testing.T) {
+	restoreEnv := initConfigTestEnv()
+	defer awstesting.PopEnv(restoreEnv)
+
+	os.Setenv("AWS_REGION", "us-east-1")
+	os.Setenv("AWS_SHARED_CREDENTIALS_FILE", testConfigFilename)
```
Comment (jasdel) [inline config/resolve_assume_role_test.go:61]: Will be helpful to rebase this PR on top of the ci with windows change to make sure this works for windows.

File: config/resolve_assume_role_test.go [hunk 2]
```diff
@@ -0,0 +1,201 @@
...
+			Body: ioutil.NopCloser(bytes.NewReader([]byte(fmt.Sprintf(
+				assumeRoleRespMsg,
+				time.Now().Add(15*time.Minute).Format("2006-01-02T15:04:05Z"))))),
+		}, nil
+	})
+
+	config, err := LoadDefaultConfig(WithHTTPClient{client})
+	// TODO: Set Assume Role Duration
+	//	AssumeRoleDuration: 30 * time.Minute,
+	if err != nil {
```
Comment (jasdel) [inline config/resolve_assume_role_test.go:181]: todo still valid?

File: config/resolve_credentials_test.go [hunk 3]
```diff
@@ -0,0 +1,222 @@
...
+
+	"github.com/aws/aws-sdk-go-v2/internal/awstesting"
+	"github.com/aws/aws-sdk-go-v2/service/sts"
+	"github.com/awslabs/smithy-go/middleware"
+	smithyhttp "github.com/awslabs/smithy-go/transport/http"
+)
+
+func TestSharedConfigCredentialSource(t *testing.T) {
+	const configFileForWindows = "testdata/credential_source_config_for_windows"
+	const configFile = "testdata/credential_source_config"
```
Comment (jasdel) [inline config/resolve_credentials_test.go:91]: nit these paths aren't compatible with windows should use `filepath.Join` instead.

File: config/resolve_assume_role_test.go (hunk 2 above)
Comment (skmcgrail) [reply config/resolve_assume_role_test.go:181]: Yes, will fix this.

**Review by jasdel — APPROVED** [review]

**Pull request merged**